pkg/apis/pipeline/v1alpha1/type.go

```diff
@@ -0,0 +1,206 @@
+// +kubebuilder:object:root=true
+// +kubebuilder:resource:scope=Namespaced,categories=kurator-dev
+// +kubebuilder:subresource:status
+
+// Pipeline is the top-level type for Kurator CI Pipeline.
+type Pipeline struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Spec   PipelineSpec   `json:"spec"`
+	Status PipelineStatus `json:"status,omitempty"`
+}
+
+// PipelineSpec defines the desired state of a Pipeline.
+type PipelineSpec struct {
+	// Description allows an administrator to provide a description of the pipeline.
+	// +optional
+	Description string `json:"description,omitempty"`
+
+	// Tasks is an ordered list of tasks in the pipeline, containing detailed information about each task.
+	// The tasks will be executed in the order they are listed.
+	Tasks []PipelineTask `json:"tasks"`
+
+	// SharedWorkspace is the name of the PVC. If not specified, a PVC with the Pipeline's name as prefix will be created by default.
+	// If not set, Kurator will create a PVC named Pipeline.name using default config
+	// +optional
+	SharedWorkspace *string `json:"sharedWorkspace,omitempty"`
+}
+
+type PipelineTask struct {
+	// Name is the name of the task.
+	Name string `json:"name"`
+
+	// TaskTemplate allows users to select a predefined task.
+	// Users can choose a task from a set list and fill in their own parameters.
+	// +optional
+	TaskTemplate *TaskTemplate `json:"TaskTemplate,omitempty"`
+
+	// CustomTask enables defining a task directly within the CRD if TaskRef is not used.
+	// This should only be used when TaskRef is not provided.
+	// +optional
+	CustomTask *CustomTask `json:"customTask,omitempty"`
+
+	// Retries represents how many times this task should be retried in case of task failure.
+	// default values is zero.
+	// +optional
+	Retries int `json:"retries,omitempty"`
+}
+
+type PredefinedTask string
+
+const (
+	// GitClone is typically the first task in the entire pipeline.
+	// It clones the user's code repository into the workspace. This allows subsequent tasks to operate on this basis.
+	// Since the pipeline is linked to a specific repository's webhook, Kurator automatically retrieves the repository information when triggered by the webhook.
+	// Users don't need to configure additional repository information for this task, except for authentication details for private repositories.
+	// This Predefined Task is origin from https://github.com/tektoncd/catalog/tree/main/task/git-clone/0.9
+	// Here are the params that user can config:
+	// - git-secret-name: The name of the secret for Git basic authentication.
+	//   Kurator uses this Git credential to clone private repositories.
+	//   The secret is formatted as follows, and more details can be found at:
+	//   https://kurator.dev/docs/fleet-manager/pipeline/
+	//   .gitconfig example:
+	//   |   [credential "https://<hostname>"]
+	//   |       helper = store
+	//   .git-credentials example:
+	//   |   https://<user>:<pass>@<hostname>
```

Comment:
This is the right auth secret format
https://tekton.dev/docs/pipelines/auth/#configuring-authentication-for-git

Reply:
ok, this is a more generic way of writing, rather than a specific task template.